soap/client.go

```diff
@@ -96,6 +91,31 @@ func (c *Client) RoundTrip(in, out Message) error {
 	return xml.NewDecoder(resp.Body).Decode(out)
 }
 
+// RoundTrip implements the RoundTripper interface.
+func (c *Client) RoundTrip(in, out Message) error {
+	headerFunc := func(r *http.Request) {
+		ct := c.ContentType
+		if ct == "" {
+			ct = "text/xml"
+		}
+		r.Header.Set("Content-Type", ct)
+		// TODO - what should happen here? there is a test case that does send in a nil & it was triggering a panic,
```

Comment:
How was this not failing before?

Well, worst case scenario is that we cannot set a SOAPAction and the request may fail - this probably depends on the server tho, and there's all kinds of them out there. I believe for now we can keep the check and just not set the SOAPAction.